lib/resty/iprepd.lua

```diff
@@ -108,9 +121,7 @@ function _M.get_reputation(self, ip)
   local reputation = self.cache:get(ip)
 
   if not reputation then
-    local httpc = http.new()
-    httpc:set_timeout(self.timeout)
-    local resp, err = httpc:request_uri(string.format("%s/%s", self.url, ip), {
+    local resp, err = self.httpc:request_uri(string.format("%s/%s", self.url, ip), {
       method  = "GET",
       headers = self.api_key_hdr,
```

Comment:
I'm wondering if we can just adjust `keepalive_timeout` and `keepalive_pool` here.

It looks like this may already be enabled by default actually.

https://github.com/ledgetech/lua-resty-http#request_uri
https://github.com/openresty/lua-nginx-module#lua_socket_keepalive_timeout
https://github.com/openresty/lua-nginx-module#lua_socket_pool_size
https://github.com/ledgetech/lua-resty-http/blob/master/lib/resty/http.lua#L127
https://github.com/ledgetech/lua-resty-http/blob/master/lib/resty/http.lua#L919-L931

Reply:
maybe @ajvb can tcpdump this?